src/tmate.py

```diff
@@ -0,0 +1,198 @@
+    script = template.render(keys_dir=KEYS_DIR, host=str(host_ip), port=PORT)
+    WORK_DIR.mkdir(parents=True, exist_ok=True)
+    KEYS_DIR.mkdir(parents=True, exist_ok=True)
+    CREATE_KEYS_SCRIPT_PATH.write_text(script, encoding="utf-8")
+    try:
+        # B603:subprocess_without_shell_equals_true false positive
+        # see https://github.com/PyCQA/bandit/issues/333
+        subprocess.check_call(["/usr/bin/chown", "-R", "ubuntu:ubuntu", str(WORK_DIR)])  # nosec
+        CREATE_KEYS_SCRIPT_PATH.chmod(755)
+        subprocess.check_call(["/usr/bin/sudo", str(CREATE_KEYS_SCRIPT_PATH)])  # nosec
+    except subprocess.CalledProcessError as exc:
+        raise KeyInstallError from exc
+
+
+def start_daemon() -> None:
+    """Install unit files and start daemon.
+
+    Raises:
+        DaemonStartError: if there was an error starting the tmate-ssh-server docker process.
+    """
+    environment = jinja2.Environment(loader=jinja2.FileSystemLoader("templates"), autoescape=True)
+    service_content = environment.get_template("tmate-ssh-server.service.j2").render(
+        WORKDIR=WORK_DIR,
+        KEYS_DIR=KEYS_DIR,
+        PORT=PORT,
+    )
+    TMATE_SSH_SERVER_SERVICE_PATH.write_text(service_content, encoding="utf-8")
+    try:
+        # B603:subprocess_without_shell_equals_true false positive
+        # see https://github.com/PyCQA/bandit/issues/333
+        subprocess.check_call(["/usr/bin/systemctl", "daemon-reload"])  # nosec
+        subprocess.check_call(["/usr/bin/systemctl", "restart", "tmate-ssh-server"])  # nosec
+        subprocess.check_call(["/usr/bin/systemctl", "enable", "tmate-ssh-server"])  # nosec
+    except subprocess.CalledProcessError as exc:
+        raise DaemonStartError from exc
+
+
+@dataclasses.dataclass
+class Fingerprints:
+    """The public key fingerprints.
+
+    Attributes:
+        rsa: The RSA public key fingerprint.
+        ed25519: The ed25519 public key fingerprint.
+    """
+
+    rsa: str
+    ed25519: str
+
+
+def get_fingerprints() -> Fingerprints:
+    """Get fingerprint from generated keys.
+
+    Raises:
+        IncompleteInitError: if the keys have not been generated by the create_keys.sh script.
+        KeyInstallError: if there was something wrong generating fingerprints from public keys.
+
+    Returns:
+        The generated public key fingerprints.
+    """
+    if not KEYS_DIR.exists() or not RSA_PUB_KEY_PATH.exists() or not ED25519_PUB_KEY_PATH.exists():
+        raise IncompleteInitError("Missing keys path(s).")
+    try:
+        # B603:subprocess_without_shell_equals_true false positive
+        # see https://github.com/PyCQA/bandit/issues/333
+        rsa_stdout = str(
+            subprocess.check_output(  # nosec
+                ["/usr/bin/ssh-keygen", "-l", "-E", "SHA256", "-f", str(RSA_PUB_KEY_PATH)]
+            ),  # the output consists of <bit-len> SHA256:<fingerprint> <user-hostname> <algorithm>
+            encoding="utf-8",
```

Comment:
From [previous PR](https://github.com/canonical/tmate-ssh-server-operator/pull/2)
[@jdkandersson] Perhaps a comment explaining what this is intended to achieve?